Source/UI/Coordinators/PasswordCoordinator.swift

```diff
@@ -194,4 +239,66 @@ extension PasswordCoordinator {
         }
         self.navigationController.pushViewController(viewController, animated: true)
     }
+    private func getPasswordFromKeychain(for identifier: Identifier) -> String? {
+        guard #available(iOS 11.3, *) else {
+            // Fallback to passsword login
+            return nil
+        }
+        var query = [String: Any]()
+        query[kSecClass as String] = kSecClassGenericPassword
+        query[kSecReturnData as String] = kCFBooleanTrue
+        query[kSecAttrAccount as String] = identifier.normalizedString as CFString
+        query[kSecAttrLabel as String] = "com.schibsted.account.biometrics.secrets" as CFString
+        query[kSecUseOperationPrompt as String] = "Please put your fingers on that button" as CFString
+
+        var queryResult: AnyObject?
+        let status = withUnsafeMutablePointer(to: &queryResult) {
+            SecItemCopyMatching(query as CFDictionary, UnsafeMutablePointer($0))
+        }
+
+        if status == noErr, let qresult =  queryResult as? Data {
+            let password = String(data: qresult as Data, encoding: .utf8)!
```

Comment:
If the data is corrupted for any reason this would crash. I'd suggest you hoist the `let password = blah` part in to the `if` (variables unwrapped in an if are usable in the same if)